### ISSUE TITLE
Wifi does not automatically reconnect if connectivity is lost

### ISSUE THREAD

**Issue by ElizabethGreene** [ElizabethGreene/GarageDoorButton#1]
REPRO STEPS:
Boot Esp8266, connect to wifi, and confirm proper operation
Reboot AP
Connectivity is lost.

**Comment by ElizabethGreene** [comment]
Got a repro leaving device and up on serial monitor overnight.  Device appears to be resetting for software watchdog.

```
........
Connected to TheGarage
IP address: 192.168.1.78
MDNS responder started
HTTP server started

 ets Jan  8 2013,rst cause:4, boot mode:(3,6)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v951aeffa
~ld

```

**Comment by ElizabethGreene** [comment]
This was found to be a memory leak caused by a too tight main loop.  Fixed in 6d3af9b.
